macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,278 @@
+      - id: sample_size
+        -orig-id: sampleSize
+        type: u2
+        doc: number of bits per sample
+      - id: reserved
+        -orig-id: futureUse1, futureUse2, futureUse3, futureUse4
+        size: 14
+        doc: reserved
+      - id: sample_area
+        -orig-id: sampleArea
+        size: num_frames * num_channels * sample_size / 8
+        doc: sampled-sound data
+  compressed_sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or NIL if samples follow data structure)
+      - id: num_channels
+        -orig-id: numChannels
+        type: u4
+        doc: number of channels in sample
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample’s encoding option
+      - id: base_frequency
+        type: u1
+        doc: base frequency of sample
+        -orig-id: baseFrequency
+      - id: num_frames
+        type: u4
+      - id: aiff_sample_rate
+        size: 10
+        doc: rate of original sample (Extended80)
+      - id: marker_chunk
+        -orig-id: markerChunk
+        type: u4
+        doc: reserved
+      - id: format
+        size: 4
+        type: str
+        encoding: ASCII
+        doc: data format type
+      - id: reserved
+        -orig-id: futureUse2
+        size: 4
+        doc: reserved
+      - id: state_vars_ptr
+        -orig-id: stateVars
+        type: u4
+        doc: pointer to StateBlock
+      - id: left_over_samples_ptr
+        -orig-id: leftOverSamples
+        type: u4
+        doc: pointer to LeftOverBlock
+      - id: compression_id
+        -orig-id: compressionID
+        type: u2
+        doc: ID of compression algorithm
+      - id: packet_size
+        -orig-id: packetSize
+        type: u2
+        doc: number of bits per packet
+      - id: snth_id
+        -orig-id: snthID
+        type: u2
+        doc: unused
+      - id: sample_size
+        -orig-id: sampleSize
+        type: u2
+        doc: number of bits per sample
+      - id: sample_area
+        -orig-id: sampleArea
+        size-eos: true
+        doc: compressed sound data
+  unsigned_fixed_point:
+    seq:
+      - id: integer_part
+        type: u2
+        doc: Integer Part of a fixed point value
+      - id: fraction_part
+        type: u2
+        doc: Fraction Part of a fixed point value
+    instances:
+      value:
+        value: integer_part + fraction_part/65535.0
+
+enums:
+  data_type:
+    0x01: square_wave_synth
+    0x03: wave_table_synth
+    0x05: sampled_synth
+  cmd_type:
+    00: null_cmd
+    03: quiet_cmd
+    04: flush_cmd
```

Comment:
Be really careful with prefixing the decimal values with padding zeros. This actually **changes the** number's **meaning**: the number literal is interpreted in **octal** notation (base 8) instead of **decimal** (base 10) if it starts with additional 0s.

For example, try the following YAML snippet in a [YAML parser](https://yaml-online-parser.appspot.com/):

```yaml
012: foo
027: bar
071: baz
```

You'll get the following output:

```js
{
  "10": "foo", // 8 * `1` + `2`
  "23": "bar", // 8 * `2` + `7`
  "57": "baz" //  8 * `7` + `1`
}
```

Here you're only lucky that the numbers `00`, `03` and `04` are written in octal base the same as in decimal. However, as you can see above, this can really cause unexpected results. Please rather eliminate the padding zeros to avoid confusion:

```suggestion
    0: null_cmd
    3: quiet_cmd
    4: flush_cmd
```